DetDescr/include/DetDescr/HcalGeometry.h

```diff
@@ -0,0 +1,152 @@
+/**
+ * @file HcalGeometry.h
+ * @brief Class that translates HCal ID into positions of strip hits
+ */
+
+#ifndef DETDESCR_HCALGEOMETRY_H_
+#define DETDESCR_HCALGEOMETRY_H_
+
+// LDMX
+#include "DetDescr/HcalID.h"
+#include "Framework/ConditionsObject.h"
+#include "Framework/Configure/Parameters.h"
+#include "Framework/Exception/Exception.h"
+
+// ROOT
+#include "TVector3.h"
+
+// STL
+#include <map>
+
+namespace hcal {
+class HcalGeometryProvider;
+}
+
+namespace ldmx {
+
+class HcalGeometryProvider;
+
+/**
+ * @class HcalGeometry
+ * @brief Implementation of HCal strip readout
+ *
+ */
+class HcalGeometry : public framework::ConditionsObject {
+ public:
+  /**
+   * Conditions object:
+   *  The name of the python configuration calling this class
+   * (Hcal/python/HcalGeometry.py) needs to match the CONDITIONS_OBJECT_NAME
+   * exactly.
+   */
+  static constexpr const char* CONDITIONS_OBJECT_NAME{"HcalGeometry"};
+
+  /**
+   * Class destructor.
+   *
+   * Does nothing because the stl containers clean up automatically.
+   */
+  virtual ~HcalGeometry() {}
+
+  /**
+   * Get a strip center position from a combined hcal ID.
+   *
+   * @throw std::out_of_range if HcalID is not on map.
+   *
+   * @param HcalID
+   * @return A TVector3 with the X, Y and Z position of the center of the bar.
+   */
+  TVector3 getStripCenterPosition(ldmx::HcalID id) const {
+    return stripPositionMap_.at(id);
+  }
+
+  /**
+   * Get the half total width for a given section(strip) for back(side) Hcal.
+   * @param section
+   * @return half total width [mm]
+   */
+  double getHalfTotalWidth(int isection) const {
+    return HalfTotalWidth_.at(isection);
+  }
+
+  /**
+   * Get the number of sections.
+   */
+  int getNumSections() const { return NumSections_; }
+
+  /**
+   * Get the number of layers for that section.
+   */
+  int getNumLayers(int isection) const { return NumLayers_.at(isection); }
+
+  /**
+   * Get the number of strips per layer for that section.
+   */
+  int getNumStrips(int isection) const { return NumStrips_.at(isection); }
+
+  static HcalGeometry* debugMake(const framework::config::Parameters& p) {
+    return new HcalGeometry(p);
+  }
```

Comment:
For Ecal, we have this for some drawing mechanisms inside of the Ecal module. Do you use this anywhere? If not, you should delete it so that it isn't accidentally used by a developer down the line. If you do use it, put in a documentation comment and add a warning that this shouldn't be used inside of processors! (Use conditions instead).

Reply:
I do not use it so I [deleted for now](https://github.com/LDMX-Software/ldmx-sw/pull/958/commits/8cd3c51eada7122290b184f04356d461949ca0fe).